src/Uno.UI/UI/Xaml/Window.Android.cs

```diff
@@ -161,6 +231,57 @@ internal void DisplayFullscreen(UIElement element)
 				_fullWindow.Child = element;
 			}
 		}
+
+		#region StatusBar properties
+		private bool IsStatusBarVisible()
+		{
+				var decorView = (ContextHelper.Current as Activity)?.Window?.DecorView;
+
+				if (decorView == null)
+				{
+					throw new global::System.Exception("Cannot check NavigationBar visibility property. DecorView is not defined yet.");
+				}
+
+				return ((int)decorView.SystemUiVisibility & (int)SystemUiFlags.Fullscreen) == 0;
+		}
+
+		public bool IsStatusBarTranslucent()
+		{
+			if (!(ContextHelper.Current is Activity activity))
+			{
+				throw new Exception("Cannot check NavigationBar translucent property. Activity is not defined yet.");
+			}
+
+			return activity.Window.Attributes.Flags.HasFlag(WindowManagerFlags.TranslucentStatus)
+				|| activity.Window.Attributes.Flags.HasFlag(WindowManagerFlags.LayoutNoLimits); ;
+		}
+		#endregion
+
+		#region NavigationBar properties
+		private bool IsNavigationBarVisible()
+		{
+			var decorView = (ContextHelper.Current as Activity)?.Window?.DecorView;
+
+			if (decorView == null)
+			{
+				throw new global::System.Exception("Cannot check NavigationBar visibility property. DecorView is not defined yet.");
+			}
+
+			return ((int)decorView.SystemUiVisibility & (int)SystemUiFlags.HideNavigation) == 0
+				|| ((int)decorView.SystemUiVisibility & (int)SystemUiFlags.LayoutHideNavigation) == 0;
+		}
+
+		private bool IsNavigationBarTranslucent()
+		{
+			if (!(ContextHelper.Current is Activity activity))
+			{
+				throw new Exception("Cannot check NavigationBar translucent property. Activity is not defined yet.");
+			}
+
+			return activity.Window.Attributes.Flags.HasFlag(WindowManagerFlags.TranslucentNavigation)
```

Comment:
Store the flags in a variable to avoid calling `activity.Window.Attributes.Flags` multiple times.